Settings for weekplanner feature 857

Taoshix opened the pull request:
# Description

Added Number of Actitivies to Display, Show Only Actitivies and Show Settings for Citizens to the web api and the settings table in the database to be used in the weekplanner and api client

Fixes related to #857 in weekplanner

## Type of change
*Delete unchecked boxes (only for Type of change)*

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

This has been tested by having the api client and weekplaner run tests that make requests to the web api to both get and insert data into the database. Specifically inputted and extracted data regarding the newly created attributes.

**Development Configuration**
*Type "flutter --version" and "dart --version" in your CMD to check versions.*

* Flutter version: 2.0.5
* Dart version: 2.12.3

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas, if necessary
- [ ] I have made corresponding changes to the documentation, if necessary
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works, if necessary
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [ ] I have Acceptance Tested this on an iOS device
- [x] I have Acceptance Tested this on an Android device

Chris31415926 committed:
Added two new entries to the database

Added NrOfActivitiesToDisplay and ShowOnlyActivities to the settings table in the database.

Chris31415926 committed:
Revert "Added two new entries to the database"

This reverts commit d814e05a51a1f539830b00c2f5d9291d518a42ce.

Chris31415926 committed:
Revert "Revert "Added two new entries to the database""

This reverts commit 538953d77657f16a74ea0d94c2ca563b8c1ec69b.

Chris31415926 committed:
Revert "Revert "Revert "Added two new entries to the database"""

This reverts commit cfa11a76588dfa9a5646cfc3de4b8eaf10c8ee93.

Chris31415926 committed:
Migrated nrOfActivitiesToDisplay and showOnlyActivities into settings table

Chris31415926 committed:
Added new item to settings table in database

Has added a boolean called ShowSettingsForCitizen that is used to indicate whether a citizen shall see their own settings tab.

makelblack159 committed:
Merge branch 'develop' into Settings-for-weekplanner-feature-857

Taoshix committed:
Merge branch 'develop' into Settings-for-weekplanner-feature-857

NikolajKrogh commented on GirafRest/Migrations/20221101082038_AddedShowOnlyActivitiesAndNrOfActivitiesToDisplayToSettingsTable.cs:
```diff
@@ -0,0 +1,36 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
```
Hvorfor har i tilføjet #nullable disable?

Taoshix replied:
Dette er auto generated fra migration tool

Jaffenheimer reviewed (changes requested):
AddedShowOnlyActivitiesAndNrOfActivitiesToDisplayToSettingsTable.Designer.cs, AddedShowOnlyActivitiesAndNrOfActivitiesToDisplayToSettingsTable.cs og AddedShowSettingsForCitizenToSettingsTable.Designer.cs er efter min mening alt alt alt for lange klasse navne og bør ændres til noget mere kort og præcist. 

Resterende kommentarer til specifik kode kan ses i @NikolajKrogh's review.

Jaffenheimer commented:
Har i testet dette også? Synes ikke jeg kan se at i har tilføjet nogle tests, bliver det i har lavet om allerede testet i forvejen?

Taoshix committed:
Merge branch 'develop' into Settings-for-weekplanner-feature-857

Taoshix committed:
Added tests for our new settings

Taoshix committed:
Fix errors

Taoshix committed:
Remove breaking change

Chris31415926 committed:
Changed class names

Changed AddedShowOnlyActivitiesAndNrOfActivitiesToDisplayToSettingsTable to AddedActivityVariablesToDatabase and  changed AddedShowSettingsForCitizenToSettingsTable to AddedSettingsFlagToDatabase

Chris31415926 committed:
Changed a class name + minor changes to former commit

Changed AddedShowPopupColumnToSettingsTable to AddedPopupFlagToDB

Jaffenheimer reviewed (approved):
This looks good to me now

AOsnes commented on GirafRest/appsettings.Development.json:
```diff
@@ -1,6 +1,6 @@
 ﻿{
     "ConnectionStrings": {
-        "DefaultConnection": "server=192.38.56.153;port=3306;userid=root;password=Giraf123;database=giraf;Allow User Variables=True"
+        "DefaultConnection": "server=localhost;port=3306;userid=root;password=Giraf123;database=giraf;Allow User Variables=True"
```
Remove localhost from commmit

Taoshix replied:
Removed localhost now

AOsnes reviewed (changes requested):
Looks good except for the comment left

Taoshix committed:
Remove localhost

Taoshix committed:
Merge branch 'develop' into Settings-for-weekplanner-feature-857

AOsnes commented:
TestUserCanCheckMultipleSettings integration test fails when running locally.

Taoshix commented:
> TestUserCanCheckMultipleSettings integration test fails when running locally.

drop the giraf database then run `dotnet-ef database update`

There was a merge conflict in the last commit

AOsnes reviewed (approved):
lgtm
